lib/iris/cube.py

```diff
@@ -2263,8 +2181,18 @@ def new_cell_measure_dims(cm_):
             if ma.count_masked(data) == 0:
                 data = data.filled()
 
+        # XXX: Slicing a single item from a masked array that is masked,
+        #      results in numpy (v1.11.1) *always* returning a MaskedConstant
+        #      with a dtype of float64, regardless of the original masked
+        #      array dtype!
+        if isinstance(data, ma.core.MaskedConstant) and \
+                data.dtype != cube_data.dtype:
+            data = ma.array(data.data, mask=data.mask, dtype=cube_data.dtype)
+
         # Make the new cube slice
-        cube = Cube(data)
+        cube = Cube(data,
+                    fill_value=self.fill_value,
+                    dtype=self._data_manager.dtype)
```

Comment:
Note that, previously the `fill_value` and `dtype` were being lost whenever we sliced a cube ... test coverage!